client/src/components/boxes/FindBoxForm/FindBoxForm.tsx

```diff
@@ -0,0 +1,167 @@
+import { FormButton, FormWrapper, FormInput, FormSelect } from '@/components';
+import { Typography, Space, Button } from 'antd';
+import { Content } from 'antd/lib/layout/layout';
+import s from './FindBoxForm.module.less';
+import { useEffect, useState } from 'react';
+import { useMutation } from '@tanstack/react-query';
+import {
+  useBoxContext,
+  boxResponse,
+  setStationAvailable,
+  APIManager,
+  fetcher,
+  FormMessage,
+  ID_NUMBER_REQUIRED,
+  TYPE_OF_BOX_REQUIRED,
+  recognizeError,
+  isFailedFetched,
+  openNotification,
+  NO_CONNECT_WITH_SERVER,
+  useAuthContext,
+  setStationUnknown,
+  getID,
+} from '@/utils';
+import { Spinner } from '@components/Layout/Spinner/Spinner';
+import { useNavigate } from 'react-router-dom';
+import { useForm } from 'antd/es/form/Form';
+
+const { Text } = Typography;
+
+const options = [
+  {
+    value: 0,
+    label: 'Puszka Wolontariusza',
+  },
+  {
+    value: 10000,
+    label: 'Puszka Stacjonarna',
+  },
+  {
+    value: 20000,
+    label: 'Puszka Firmowa',
+  },
+];
+
+type FormInput = {
+  id_number: string;
+  box_type: 0 | 10000 | 20000;
+};
+
+export const FindBoxForm = () => {
+  const navigate = useNavigate();
+  const { username } = useAuthContext();
+  if (username === null) {
+    navigate('/liczymy/login');
+    return null;
+  } else {
+    const [message, setMessage] = useState<FormMessage | undefined>();
+    const [form] = useForm();
+    const { createBox } = useBoxContext();
+
+    setStationAvailable();
+
+    const mutation = useMutation<boxResponse, unknown, number, unknown>({
+      mutationFn: (volunteerId: number) => fetcher(APIManager.findBoxURL(volunteerId)),
+      onError: (error) => {
+        setMessage({ type: 'error', content: recognizeError(error) });
+      },
+      onSuccess: (data) => {
+        setMessage({
+          type: 'success',
+          content: `Pomyślnie znaleziono puszkę dla identyfikatora: ${data.collectorIdentifier}`,
+        });
+        createBox(
+          [data.collector.firstName, data.collector.lastName].join(' '),
+          data.collectorIdentifier,
+          data.id.toString(),
+        );
+        form.resetFields();
+        // TODO: Probalby can get rid of setTimeout
+        setTimeout(() => navigate('/liczymy/boxes/settle/2'), 1000);
```

Comment:
To powinno się dziać od razu, zgadzam się z todo.

Reply:
Message powinien być widoczny też na następnej stronie.

Reply:
Tak, powinno działać od razu lecz po przeklikiwaniu parę razy wystąpiła sytuacja że nie działało bez tego i na następnej stronie odczytywało raz na jakiś czas że w contextcie dane są równe null, poszukam na to rozwiązania